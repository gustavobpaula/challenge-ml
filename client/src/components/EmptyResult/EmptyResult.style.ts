import styled from '@emotion/styled'

export const Title = styled.h1`
  color: ${({ theme }) => theme.colors.gray1};
  font-size: ${({ theme }) => theme.type.sizes.lg};
  font-weight: 600;
`

export const ToDo = styled.li`
  color: ${({ theme }) => theme.colors.gray2};
  font-size: ${({ theme }) => theme.type.sizes.sm};
  line-height: 150%;
`

export const Content = styled.div`
  padding: ${({ theme }) => theme.spacings.rule3};
`
