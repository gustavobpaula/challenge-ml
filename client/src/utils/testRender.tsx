import React from 'react'
import { render, RenderOptions } from '@testing-library/react'
import { ThemeProvider } from '@emotion/react'

import theme from 'config/theme'

type RenderComponentProps = {
  component: React.ReactNode
  useTheme?: boolean
  options?: RenderOptions
}

type ProviderProps = {
  children: React.ReactNode
}

const Theme = ({ children }: ProviderProps) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
)

export const RenderComponent = ({
  component,
  useTheme = false,
  options = {},
}: RenderComponentProps) => {
  const providers = [useTheme && Theme]

  const componentWithProviders = providers.reduce((accComponent, Provider) => {
    if (!Provider) return accComponent

    return <Provider>{accComponent}</Provider>
  }, component)

  return render(<>{componentWithProviders}</>, options)
}
