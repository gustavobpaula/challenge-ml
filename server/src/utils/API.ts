import fetch from 'node-fetch'
import { URLSearchParams } from 'url'

const ML_API = 'https://api.mercadolibre.com'

const API = {
  /**
   * Return currency object
   */
  getCurrency: async (id: string) => {
    try {
      const response = await fetch(`${ML_API}/currencies/${id}`)
      return await response.json()
    } catch (error) {
      throw new Error(error)
    }
  },
  /**
   * Return items by query
   */
  search: async (query: string) => {
    const params = new URLSearchParams()
    params.append('q', query)

    try {
      const response = await fetch(
        `${ML_API}/sites/MLA/search?${params.toString()}`
      )

      return await response.json()
    } catch (error) {
      throw new Error(error)
    }
  },
}

export default API
