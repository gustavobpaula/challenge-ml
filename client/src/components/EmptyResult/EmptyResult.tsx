import * as S from './EmptyResult.style'

function EmptyResult() {
  return (
    <S.Content>
      <S.Title>No hay publicaciones que coincidan con tu búsqueda.</S.Title>
      <ul>
        <S.ToDo>
          <strong>Revisá la ortografía</strong> de la palabra.
        </S.ToDo>
        <S.ToDo>
          Utilizá <strong>palabras más genéricas</strong> o menos palabras.
        </S.ToDo>
      </ul>
    </S.Content>
  )
}

export default EmptyResult
