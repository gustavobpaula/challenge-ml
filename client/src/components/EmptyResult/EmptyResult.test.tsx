import { screen } from '@testing-library/react'

import { RenderComponent } from 'utils'
import EmptyResult from './EmptyResult'

const makeSut = () => {
  return RenderComponent({
    component: <EmptyResult />,
    useTheme: true,
  })
}

describe('Breadcrumb component', () => {
  it('Should render a todo list', () => {
    makeSut()

    expect(screen.getByRole('list', { name: /todo list/i })).toBeInTheDocument()
  })

  it('Should render a empty result message', () => {
    makeSut()

    expect(
      screen.getByText(/No hay publicaciones que coincidan con tu b√∫squeda./i)
    ).toBeInTheDocument()
  })
})
