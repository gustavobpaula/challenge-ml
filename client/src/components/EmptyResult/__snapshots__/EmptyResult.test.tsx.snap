// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumb component Should render match snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-12128pa"
  >
    <h1
      class="css-1fe6w8t"
    >
      No hay publicaciones que coincidan con tu búsqueda.
    </h1>
    <ul
      aria-label="todo list"
    >
      <li
        class="css-uv3dt8"
      >
        <strong>
          Revisá la ortografía
        </strong>
         de la palabra.
      </li>
      <li
        class="css-uv3dt8"
      >
        Utilizá 
        <strong>
          palabras más genéricas
        </strong>
         o menos palabras.
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
